internal/controller/grafanaorganization_controller.go

```diff
@@ -160,24 +160,30 @@ func (r GrafanaOrganizationReconciler) reconcileCreate(ctx context.Context, graf
 		return ctrl.Result{}, nil
 	}
 
+	var lastError error
+
 	// Configure the shared organization in Grafana
 	if err := r.configureSharedOrg(ctx); err != nil {
-		return ctrl.Result{}, errors.WithStack(err)
+		lastError = err
 	}
 
 	// Configure the organization in Grafana
 	if err := r.configureOrganization(ctx, grafanaOrganization); err != nil {
-		return ctrl.Result{}, errors.WithStack(err)
+		lastError = err
 	}
 
 	// Update the datasources in the CR's status
 	if err := r.configureDatasources(ctx, grafanaOrganization); err != nil {
-		return ctrl.Result{}, errors.WithStack(err)
+		lastError = err
 	}
 
 	// Configure Grafana RBAC
 	if err := r.configureGrafanaSSO(ctx); err != nil {
-		return ctrl.Result{}, errors.WithStack(err)
+		lastError = err
+	}
+
+	if lastError != nil {
+		return ctrl.Result{}, errors.WithStack(lastError)
```

Comment:
Can you please use https://pkg.go.dev/errors@master#Join instead?

Reply:
They are only logged for the shared org though right?

Reply:
Because in any case, the last error is logged twice so maybe we do not need the log lines we added if the error is logged

Reply:
I think there are some improvements to be made regarding our error logging. But I would rather not do  this here.
I made sure all errors are logged when they happened.

Reply:
Sure let's fix the ongoing issue then